src/components/TextField/TextField.tsx

```diff
@@ -129,394 +120,356 @@ export type TextFieldProps = NonMutuallyExclusiveProps &
     | {disabled: true}
     | {onChange(value: string, id: string): void});
 
-type CombinedProps = TextFieldProps & WithAppProviderProps;
-
-const getUniqueID = createUniqueIDFactory('TextField');
-
-class TextField extends React.PureComponent<CombinedProps, State> {
-  static getDerivedStateFromProps(nextProps: CombinedProps, prevState: State) {
-    return {id: nextProps.id || prevState.id};
-  }
-
-  private input: HTMLElement;
-  private buttonPressTimer: number;
-  private prefix = React.createRef<HTMLDivElement>();
-  private suffix = React.createRef<HTMLDivElement>();
-
-  constructor(props: CombinedProps) {
-    super(props);
-
-    // eslint-disable-next-line react/state-in-constructor
-    this.state = {
-      height: null,
-      focus: props.focused || false,
-      id: props.id || getUniqueID(),
-    };
-  }
-
-  componentDidMount() {
-    if (!this.props.focused) {
-      return;
-    }
-
-    this.input.focus();
-  }
-
-  componentDidUpdate({focused: wasFocused}: CombinedProps) {
-    const {focused} = this.props;
-
-    if (!wasFocused && focused) {
-      this.input.focus();
-    } else if (wasFocused && !focused) {
-      this.input.blur();
+export function TextField({
+  prefix,
+  suffix,
+  placeholder,
+  value,
+  helpText,
+  label,
+  labelAction,
+  labelHidden,
+  disabled,
+  clearButton,
+  readOnly,
+  autoFocus,
+  focused,
+  multiline,
+  error,
+  connectedRight,
+  connectedLeft,
+  type,
+  name,
+  id: idProp,
+  role,
+  step = 1,
+  autoComplete,
+  max = Infinity,
+  maxLength,
+  min = -Infinity,
+  minLength,
+  pattern,
+  spellCheck,
+  ariaOwns,
+  ariaControls,
+  ariaActiveDescendant,
+  ariaAutocomplete,
+  showCharacterCount,
+  align,
+  onClearButtonClick,
+  onChange,
+  onFocus,
+  onBlur,
+}: TextFieldProps) {
+  const intl = useI18n();
+  const [height, setHeight] = useState<number | null>(null);
+  const [focus, setFocus] = useState(focused || false);
+
+  const id = useUniqueId('TextField', idProp);
+
+  const inputRef = useRef<HTMLElement>(null);
+  const prefixRef = useRef<HTMLDivElement>(null);
+  const suffixRef = useRef<HTMLDivElement>(null);
+  const buttonPressTimer = useRef<number>();
+  const mounted = useMountedRef();
+
+  useEffect(() => {
+    if (!inputRef.current) return;
+    if (focused) {
+      inputRef.current.focus();
+    } else if (focused === false) {
```

Comment:
Right now if focused isn't true, it'll be false, won't it? We can probably kill this else if with a return in the if statement or an else or a ternary (focused ? ...focus() : ... blur())